Loading screen improvement

okan35 opened the pull request:
<!-- Hey there. Thank you so much for improving NewPipe, and filling out the details. Having roughly the same layout helps everyone considerably :)-->

#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
<!-- While bullet points are the norm in this section, feel free to write free-form text instead of a list -->
Currently when we refresh what's new, we see only loading screen on main screen with number of videos loading and at the same time we can see this on notification. 
I removed loading screen from main screen so while app is loading new videos, we can also browse and watch videos.We should not be waiting for videos to finish loading so we can watch video or check what we have in list of videos, this definitely was not a good idea so I changed this. 
While videos load we can still check the content. And if the someone wants to see number of videos that are loading, he/she can do so by checking the notification.



#### Fixes the following issue(s)
<!-- Also add any other links relevant to your change. -->
- Fixes #4952

#### APK testing 
<!-- Use a new, meaningfully named branch. The name is used as a suffix for the app ID to allow installing and testing multiple versions of NewPipe, e.g. "commentfix", if your PR implements a bugfix for comments. (No names like "patch-0" and "feature-1".)  -->
[app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/5719598/app-debug.zip)



#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

B0pol commented:
I really dislike this behaviour.

The loading screen is there to show the user it's loading and it's not frozen.

Now you don't know the progress, it refreshes when it's loaded and really feels "wow it didn't work, I'm going to press refresh button again" with about 50 subs.

On top of that, it seems you didn't read contribution guidelines, because you don't follow them

> Send PR that only cover one specific issue/solution/bug. Do not send PRs that are huge and consists of multiple
independent solutions

okan35 commented:
I now see where you are coming from because if you click to refresh the feed it shows like there is nothing going on. But if you pull to refresh it shows you a loading icon until the process is done. 

We can also add that small icon so people dont pull to refresh bit click to refresh can see what is going on.

As to issue part, this actually was in #3329, you can see my comment and I don't know why it was closed as the requests in that issue were not done yet and only implemented small part of it which is pull to refresh.

Also instead of opening new issue I added mine in the same issue as it is covered under improvement of refresh.

opusforlife2 commented:
@okan35 We now strictly follow a policy of "one idea per issue/PR". So you can just open a new issue, describe the problem this PR is solving, and link it to this PR.

okan35 commented:
> @okan35 We now strictly follow a policy of "one idea per issue/PR". So you can just open a new issue, describe the problem this PR is solving, and link it to this PR.

I just did that and later will add the progress dialogue  or icon whatever for click refresh, pull to refresh already does that.

opusforlife2 commented:
@okan35 You can use keywords to link a PR to an issue. https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue

okan35 commented:
> 
> 
> @okan35 You can use keywords to link a PR to an issue. https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue

Will do it next time, didn't know this existed.

okan35 commented:
Added the necessary change for showing loading icon on click to refresh. I believe the feature is done now.

Stypox commented on app/src/main/java/org/schabi/newpipe/local/feed/FeedFragment.kt:
```diff
@@ -174,40 +174,42 @@ class FeedFragment : BaseListFragment<FeedState, Unit>() {
-        animateView(refresh_root_view, false, 0)
-        animateView(items_list, false, 0)
+        animateView(refresh_root_view, true, 0)
+        animateView(items_list, true, 0)
```
There is no need for this, i.e. to show `items_list`, if you never hide it ;-)

Stypox commented on app/src/main/java/org/schabi/newpipe/local/feed/FeedFragment.kt:
```diff
@@ -280,6 +282,7 @@ class FeedFragment : BaseListFragment<FeedState, Unit>() {
 
     private fun handleErrorState(errorState: FeedState.ErrorState): Boolean {
         hideLoading()
+        swipeRefreshLayout.isRefreshing=false
```
No need for this, it is already inside `hideLoading()`

Stypox reviewed (changes requested):
I like this, though I am not sure if it is a good idea not to show any progress indicator except for the hard-to-access notification. You could show the loading bar, which previously was shown at the center of the screen, just below the `Feed last updated: ...` bar at the top. Also, please remove all comments, they are not needed anymore. Thank you for this contribution that would decrease time waste :-D

okan35 commented:
> 
> 
> I like this, though I am not sure if it is a good idea not to show any progress indicator except for the hard-to-access notification. You could show the loading bar, which previously was shown at the center of the screen, just below the `Feed last updated: ...` bar at the top. Also, please remove all comments, they are not needed anymore. Thank you for this contribution that would decrease time waste :-D

I pushed the changes for those reviews you left. I would suggest that we do that feature of showing progress of video fetching in another pull request.  But it is a good idea to show it like that, below feed last updated when user refreshes video list, and when it finishes we can hide it back,

Stypox commented on app/src/main/java/org/schabi/newpipe/local/feed/FeedFragment.kt:
```diff
@@ -340,4 +314,4 @@ class FeedFragment : BaseListFragment<FeedState, Unit>() {
             return feedFragment
         }
     }
-}
```
Add back the newline at the end of the file (checkstyle should have complained)

Stypox commented on app/src/main/java/org/schabi/newpipe/local/feed/FeedFragment.kt:
```diff
@@ -117,10 +117,6 @@ class FeedFragment : BaseListFragment<FeedState, Unit>() {
         }
     }
 
-    // /////////////////////////////////////////////////////////////////////////
```
Bring back these comments.. sorry if I was unclear, I meant that old code shouldn't be commented-out but directly removed. But useful comments like this should be kept.

Stypox reviewed (changes requested):
> I would suggest that we do that feature of showing progress of video fetching in another pull request.

Yeah, ok, but both PRs need to ship in the same release, otherwise we are taking away the progress indicator from users, which would be cumbersome. Are you willing to make another PR right after this one?

okan35 commented:
I can make another pr but I will do it around this week as it is a matter of time :D

okan35 commented:
I added that progress text below video last fetched and those comments are back as I went to that commit. Updated the apk. To me it looked nice as it shows up with progress and disappears when it is finished.

Stypox commented on app/src/main/java/org/schabi/newpipe/local/feed/FeedFragment.kt:
```diff
@@ -117,10 +110,6 @@ class FeedFragment : BaseListFragment<FeedState, Unit>() {
         }
     }
 
-    // /////////////////////////////////////////////////////////////////////////
```
Bring back these comments ;-)

Stypox reviewed (commented):
Could you rebase?

litetex commented:
Closing this for now:
* no progress
* no GitHub actions build
* merge conflicts
* Snyk Found vulnerabilities!
* No feedback from the author

Feel free to reopen it when there is progress again.
